### ISSUE TITLE
Support instance variables without defaults

### ISSUE THREAD

**Issue by hoefling** [sphinx-doc/sphinx#7051]
[PEP 526](https://www.python.org/dev/peps/pep-0526/#class-and-instance-variable-annotations) allows to define type annotations for class/instance variables. Sphinx supports vars with default values, but not without.

Consider the following example from PEP 526, enhanced with a docstring:

```python
# pep526.py

from typing import ClassVar, Dict

class BasicStarship:
    """
    An example from :pep:`526`.

    :ivar str captain: instance variable with default
    :ivar int damage: instance variable without default
    :cvar stats: class variable
    :vartype stats: dict[str, int]
    """
    captain: str = 'Picard'               # instance variable with default
    damage: int                           # instance variable without default
    stats: ClassVar[Dict[str, int]] = {}  # class variable
```
Generating docs from

```rst
``pep526`` Example
==================

.. automodule:: pep526
   :members:
   :undoc-members:
   :show-inheritance:
```

yields a warning in the terminal:

```
pep526.py:docstring of pep526.BasicStarship:: WARNING: py:obj reference target not found: damage
```

and renders the docs like this:

![image](https://user-images.githubusercontent.com/4455652/72914028-5285f980-3d3e-11ea-920b-757124b17fa5.png)

The request is thus to treat an instance var without default value same way as a regular instance var, if it's possible at all. I'm fine with the rendered result, but we rely heavily on type hints and I have to put each warning to `nitpick_ignore` as our CI pipeline builds the docs with `-Wn`. Another great disadvantage is that the generated inventory doesn't contain the attributes. In the above example, the inventory would only have

```
py:attribute
	pep526.BasicStarship.captain             api/pep526.html#pep526.BasicStarship.captain
	pep526.BasicStarship.stats               api/pep526.html#pep526.BasicStarship.stats
```
so no crossref'ing to `damage` attribute is possible.

**Comment by tk0miya** [comment]
+1: Reasonable.

**Comment by tk0miya** [comment]
Fixed by #7057 
Thank you for reporting!

**Comment by hoefling** [comment]
@tk0miya I have tried out the patches from your PR, works like a charm! Awesome, thank you for the contribution! Can't wait for the 2.4.0 release!
